test/cluster/test_hints.py

```
@@ -500,3 +604,797 @@ async def migration_reached_streaming():
+    # after changing SMP: 1 -> 3 -> 2.
+    expected_rows = await populate_segments(3)
+
+    async def restart_with_smp(server_id, smp: int, hinted_handoff_enabled: bool):
+        await manager.server_stop_gracefully(server_id)
+        await manager.server_update_cmdline(server_id, [f"--smp={smp}"])
+        hh_enabled = "true" if hinted_handoff_enabled else "fake_dc"
+        await manager.server_update_config(server_id, "hinted_handoff_enabled", hh_enabled)
+        await manager.server_start(server_id)
+
+    async def check_balanced(hints_dir, num_shards: int) -> None:
+        async def check():
```

Ruff (0.16.3)

[warning] Missing return type annotation for private function `check`

Add return type annotation: `bool | None`

(ANN202)